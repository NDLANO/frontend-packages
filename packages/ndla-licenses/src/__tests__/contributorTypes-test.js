/**
 * Copyright (c) 2017-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

/* eslint-env jest */

import {
  contributorTypes,
  contributorGroups,
  mkContributorString,
  getGroupedContributorDescriptionList,
} from '../contributorTypes';

test('ContributorTypes keys is the same for each supported language', () => {
  const enLength = Object.keys(contributorTypes.en);
  const nbLength = Object.keys(contributorTypes.nb);
  expect(enLength).toEqual(nbLength);
});

test('Each contributor type in a contributor group has a valid translation', () => {
  contributorGroups.processors.forEach(processor => {
    expect(contributorTypes.en[processor]).toBeDefined();
    expect(contributorTypes.nb[processor]).toBeDefined();
  });

  contributorGroups.creators.forEach(creator => {
    expect(contributorTypes.en[creator]).toBeDefined();
    expect(contributorTypes.nb[creator]).toBeDefined();
  });

  contributorGroups.rightsholders.forEach(rightsholder => {
    expect(contributorTypes.en[rightsholder]).toBeDefined();
    expect(contributorTypes.nb[rightsholder]).toBeDefined();
  });
});

const contributorList = [
  {
    type: 'Director',
    name: 'Francis Ford Coppola',
  },
  {
    type: 'Linguistic',
    name: 'Mario Puzo',
  },
  {
    type: 'Rightsholder',
    name: 'Paramount Pictures',
  },
  {
    type: 'Distributor',
    name: 'Alfran Productions',
  },
];

test('Makes a translated comma seperated contributor string', () => {
  const contributorStringNb = mkContributorString(contributorList, 'nb');
  expect(contributorStringNb).toMatchSnapshot();
  const contributorStringEn = mkContributorString(contributorList, 'en');
  expect(contributorStringEn).toMatchSnapshot();
});

test('Makes a translated comma separated contributor string (ignores rightsholder prefix)', () => {
  const contributorString = mkContributorString(contributorList, 'nb', 'rightsholder');
  expect(contributorString).toMatchSnapshot();
});

test('Get a grouped contributor description list for each supported language', () => {
  const copyright = {
    creators: [
      {
        type: 'Director',
        name: 'Francis Ford Coppola',
      },
    ],
    processors: [
      {
        type: 'Processor',
        name: 'Francis Ford Coppola',
      },
      {
        type: 'Linguistic',
        name: 'Mario Puzo',
      },
    ],
    rightsholders: [
      {
        type: 'Rightsholder',
        name: 'Paramount Pictures',
      },
      {
        type: 'Distributor',
        name: 'Alfran Productions',
      },
    ],
  };

  expect(getGroupedContributorDescriptionList(copyright, 'nb')).toMatchSnapshot();
  expect(getGroupedContributorDescriptionList(copyright, 'en')).toMatchSnapshot();
});
