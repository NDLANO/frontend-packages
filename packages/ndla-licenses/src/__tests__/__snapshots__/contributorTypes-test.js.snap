// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get a grouped contributor description list for each supported language 1`] = `
[
  {
    "description": "Regissør Francis Ford Coppola",
    "label": "Opphaver",
    "metaType": "author",
  },
  {
    "description": "Paramount Pictures, Distributør Alfran Productions",
    "label": "Rettighetshaver",
    "metaType": "copyrightHolder",
  },
  {
    "description": "Francis Ford Coppola, Språklig Mario Puzo",
    "label": "Bearbeider",
    "metaType": "contributor",
  },
]
`;

exports[`Get a grouped contributor description list for each supported language 2`] = `
[
  {
    "description": "Director Francis Ford Coppola",
    "label": "Originator",
    "metaType": "author",
  },
  {
    "description": "Paramount Pictures, Distributor Alfran Productions",
    "label": "Rightsholder",
    "metaType": "copyrightHolder",
  },
  {
    "description": "Francis Ford Coppola, Linguistic Mario Puzo",
    "label": "Processor",
    "metaType": "contributor",
  },
]
`;

exports[`Makes a translated comma separated contributor string (ignores rightsholder prefix) 1`] = `"Regissør Francis Ford Coppola, Språklig Mario Puzo, Paramount Pictures, Distributør Alfran Productions"`;

exports[`Makes a translated comma seperated contributor string 1`] = `"Regissør Francis Ford Coppola, Språklig Mario Puzo, Rettighetshaver Paramount Pictures, Distributør Alfran Productions"`;

exports[`Makes a translated comma seperated contributor string 2`] = `"Director Francis Ford Coppola, Linguistic Mario Puzo, Rightsholder Paramount Pictures, Distributor Alfran Productions"`;
